### PR TITLE
Retroactively fix the nextXid on a known broken timeline (v3)

### PR DESCRIPTION
**Pull request by hlinnaka**
This one particular timeline in production hit the nextXid bug. Add a one-off hack that will fix the nextXid on that particular timeline.

### CODE REVIEW

File: pageserver/src/walingest.rs [hunk 1]
```diff
@@ -109,6 +113,43 @@ impl WalIngest {
...
+        // before that happens anyway, but better safe than sorry.
+        //
+        if self.checkpoint.nextXid.value == 4294968320 && // 1::1024, the incorrect value
+            // only apply this on the one broken tenant
+            modification.tline.tenant_shard_id.tenant_id == TenantId::from_hex("df254570a4f603805528b46b0d45a76c").unwrap() &&
+            lsn < Lsn::from_str("380/00000000").unwrap() &&
+            decoded.xl_xid != pg_constants::INVALID_TRANSACTION_ID
+        {
+            self.checkpoint.nextXid = FullTransactionId {
+                value: (decoded.xl_xid + 10000) as u64,
```
Comment (VladLazar) [inline pageserver/src/walingest.rs:143]: Under the assumption that they won't reach real XID wraparound until `36D/00000000` this addition is fine.
How did you infer the assumption though?

**Review by VladLazar — APPROVED** [review]

File: pageserver/src/walingest.rs (hunk 1 above)
Comment (kelvich) [reply pageserver/src/walingest.rs:143]: fyi: https://neondb.slack.com/archives/C06F5UJH601/p1706605368569849

Comment (hlinnaka) [reply pageserver/src/walingest.rs:143]: > Under the assumption that they won't reach real XID wraparound until `36D/00000000` this addition is fine.
> How did you infer the assumption though?

They are somewhat above 2 billion XIDs at the moment, and there's in the ballpark of 5GB of LSN space between the current end of the branch and `36D/00000000`. Every consumed XID is WAL logged one way or another, so it's impossible to consume another 2 billion transactions in < 5 GB of WAL.

**Comment by hlinnaka** [comment]
This was handled as a one-off thing, closing

**Pull request closed without merging**